test/rb/generate_data.py

```diff
@@ -0,0 +1,598 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019, 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Generate data for rb fitters tests
+"""
+
+import os
+import sys
+import json
+from test.utils import save_results_as_json, convert_ndarray_to_list_in_data
+import numpy as np
+import qiskit
+import qiskit.ignis.verification.randomized_benchmarking as rb
+from qiskit.ignis.verification.randomized_benchmarking import \
+    RBFitter, InterleavedRBFitter, PurityRBFitter, CNOTDihedralRBFitter
+from qiskit.providers.aer.noise import NoiseModel
+from qiskit.providers.aer.noise.errors.standard_errors import depolarizing_error,\
+    thermal_relaxation_error, coherent_unitary_error
+
+# fixed seed for simulations and for rb random seed
+SEED = 42
+
+
+def rb_circuit_execution(rb_opts: dict, shots: int):
+    """
+    Create rb circuits with depolarizing error and simulates them
+
+    Args:
+        rb_opts: the options for the rb circuits
+        shots: number of shots for each circuit simulation
+
+    Returns:
+        list: list of Results of the circuits simulations
+        list: the xdata of the rb circuit
+
+    """
+    # Load simulator
+    backend = qiskit.Aer.get_backend('qasm_simulator')
+    basis_gates = ['u1', 'u2', 'u3', 'cx']
+
+    rb_circs, xdata = rb.randomized_benchmarking_seq(**rb_opts)
+
+    noise_model = NoiseModel()
```

Comment:
This code (lines 49-50, 54-59) seem to be duplicated in most of the generation functions - maybe it's better to define it in one function and call it once?